test/integration/src/test/java/com/liferay/faces/portal/test/integration/demo/PrimeFacesUsersPortletTester.java

```diff
@@ -48,506 +52,621 @@
-		browserDriver.waitFor(ExpectedConditions.stalenessOf(imgWebElement));
-		browserStateAsserter.assertElementDisplayed(portraitXpath);
-		imgWebElement = browserDriver.findElementByXpath(portraitXpath);
+			// 3. Click the *Preferences* link to navigate to edit mode.
+			String preferencesLinkXpath = "//a[contains(@id,':preferences')]";
+			browserDriver.clickElement(preferencesLinkXpath);
+
+			// 4. Enter "5" for the number of rows-per-page so that the pagination can be better tested with a small
+			// data set.
+			String rowsPerPageInputXpath = "//input[contains(@id,':rowsPerPage')]";
+			browserDriver.waitForElementEnabled(rowsPerPageInputXpath);
+			browserDriver.clearElement(rowsPerPageInputXpath);
+			browserDriver.sendKeysToElement(rowsPerPageInputXpath, "5");
 
-		// Verify that the height of the displayed photo is different than the height noted earlier.
-		Dimension imageSizeAfter = imgWebElement.getSize();
-		Assert.assertNotEquals(imageSizeBefore.getHeight(), imageSizeAfter.getHeight());
+			// 5. Click the *Submit* button to save the rows-per-page preference.
+			browserDriver.clickElement("//input[contains(@id,':submit')]");
+			browserDriver.waitForElementDisplayed(SCREEN_NAME_CELL_XPATH);
+			browserDriver.setWaitTimeOut(TestUtil.getBrowserDriverWaitTimeOut());
+			isStateReset = true;
+		}
+	}
+
+	private boolean areImageSizesEqual(Dimension expectedDimensions, Dimension actualDimensions) {
+
+		// Image sizes may be rounded from floating point values. So instead of verifying that the sizes are exactly
+		// equal, verify that the actual size is one less than, one greater than, or exactly the same as the expected
+		// size.
+		int expectedHeight = expectedDimensions.getHeight();
+		int expectedWidth = expectedDimensions.getWidth();
+
+		int expectedHeightMin = expectedHeight - 1;
+		int expectedWidthMin = expectedWidth - 1;
+		int expectedHeightMax = expectedHeight + 1;
+		int expectedWidthMax = expectedWidth + 1;
+		int actualHeight = actualDimensions.getHeight();
+		int actualWidth = actualDimensions.getWidth();
+
+		return ((expectedHeightMin <= actualHeight) && (actualHeight <= expectedHeightMax)) &&
+			((expectedWidthMin <= actualWidth) && (actualWidth <= expectedWidthMax));
 	}
 
 	/**
 	 * Extract the column values from the dataTable associated with the specified column name.
 	 */
 	private List<String> extractColumnValuesFromDataTable(BrowserDriver browserDriver, String columnName) {
 
-		// While the "Last Page" icon is still active:
+		// While the *Last Page* button is still active:
 		List<String> columnValues = new ArrayList<String>();
 		String columnElementsXpath = "//span[contains(@id,':" + columnName + "Cell')]";
+		WebDriver webDriver = browserDriver.getWebDriver();
 
-		while (browserDriver.findElementsByXpath(
-					"//a[contains(@class,'ui-state-disabled') and contains(@aria-label,'Last Page')]").isEmpty()) {
+		while (!navigationButtonClassDisabled("Last").apply(webDriver)) {
```

Comment:
@stiemannkj1 I know it would be really long, but I think that `navigationButtonClassDisabledCondition` would read better, because it would read like "condition.apply" -- right now it reads like "disabled.apply" like it is supposed to disable it or something.